app/views/site/_selection_submit.html.erb

```diff
@@ -0,0 +1,25 @@
+<script type="text/javascript">
+$('#selection-submit').click(function(){
+    var needText = $('.need-text');
+    var numFields = needText.toArray().length;
+    var values = [];
+    var cont = true;
+    for(i = 0; i < numFields; i++){
+        var text = needText.eq(i).val();
+        values.push(text);
+        if(text === ""){cont = false}
+    }
```

Comment:
You shouldn't be making a model query directly in a form like this - if you need these values, load them in the corresponding controller action into a variable to access here.